prime-router/src/main/kotlin/LookupTable.kt

```diff
@@ -164,6 +194,22 @@ class LookupTable(
             .map { row -> row[selectColumnNumber] }
     }
 
+    fun filter(
+        filters: Map<String, String>,
+        ignoreCase: Boolean = true
+    ): LookupTable {
+        if (filters.isEmpty() || filters.count() == 0) return this
+        val filteredRows = table
+            .filter { row ->
+                filters.all { (k, v) ->
+                    val filterColumNumber = headerIndex[k.lowercase()] ?: error("$k doesn't exist in lookup table")
+                    row[filterColumNumber].equals(v, ignoreCase)
+                }
+            }
+        val headerAndFilteredRows = listOf(headerRow) + filteredRows
+        return LookupTable(headerAndFilteredRows)
+    }
+
     /**
```

Comment:
Creates the method that filters the table and then returns a whole new LookupTable object to operate on